backend/src/controller/chatController.ts

```diff
@@ -48,19 +53,20 @@ async function handleHigherLevelChat(
 	chatModel: ChatModel
 ) {
 	// transform the message according to active defences
-	chatResponse.transformedMessage = transformMessage(
+	const transformedMessage = transformMessage(
 		message,
 		req.session.levelState[currentLevel].defences
 	);
-	// if message has been transformed then add the original to chat history and send transformed to chatGPT
-	const messageIsTransformed = chatResponse.transformedMessage !== message;
-	if (messageIsTransformed) {
+	if (transformedMessage) {
+		chatResponse.transformedMessage = transformedMessage;
+		// if message has been transformed then add the original to chat history and send transformed to chatGPT
 		req.session.levelState[currentLevel].chatHistory.push({
 			completion: null,
 			chatMessageType: CHAT_MESSAGE_TYPE.USER,
 			infoMessage: message,
 		});
```

Comment:
I've kept the existing functionality here... but I'm thinking we should add the transformed message to the chat history. What do we think? 🤔

Reply:
Ah so i've worded that comment badly here (was confused myself for a while), i meant that the original message is added as an infoMessage and not a completion, which would mean it does not get sent to openAI but can still access it in the frontend chat. 
The transformed message is still added to the chatHistory as a completion, so it will be the one that is sent to openAI 

![image](https://github.com/ScottLogic/prompt-injection/assets/118981273/e706d75d-e301-4fdf-828c-357835b51b48)

Reply:
ahh I see, thanks for looking into that